### PR TITLE
Upstream 26.04.21

### PR DESCRIPTION
**Pull request by RassK**
### Actually build profiler in release mode [(#1362)](https://github.com/DataDog/dd-trace-dotnet/pull/1362)

Actually build the profiler in release mode. The changes in #1356 attempted to do this, but placed the build configuration changes into the display name of the tasks, not the actual command.

### Bump OpenTracing from 0.12.0 to 0.12.1 in /src/Datadog.Trace.OpenTracing [(#1385)](https://github.com/DataDog/dd-trace-dotnet/pull/1385)

Bumps [OpenTracing](https://github.com/opentracing/opentracing-csharp) from 0.12.0 to 0.12.1.
- [Release notes](https://github.com/opentracing/opentracing-csharp/releases)
- [Commits](https://github.com/opentracing/opentracing-csharp/compare/0.12.0...0.12.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

### Update dependabot all in one go [(#1387)](https://github.com/DataDog/dd-trace-dotnet/pull/1387)

- For Datadog.Trace, allow all dependencies. For other projects, only allow direct dependencies. This cuts down on noise since all other projects go through Datadog.Trace
- For Datadog.Trace and Datadog.Trace.ClrProfiler.Managed, add packages that should be ignored

### Update dependabot to manually ignore dependencies [(#1399)](https://github.com/DataDog/dd-trace-dotnet/pull/1399)

Copy/paste the Datadog.Trace.csproj ignored dependencies everywhere, since Datadog.Trace triggers issues everywhere. While potentially noisier than a small "allow" list, this will raise PR's if we ever add new packages which is better than not raising PR's.

### Add symbols to MSI [(#1364)](https://github.com/DataDog/dd-trace-dotnet/pull/1364)

* Add symbols to MSI
* Update the PrepareRelease tool

### Include the job attempt number in integration test log uploads [(#1403)](https://github.com/DataDog/dd-trace-dotnet/pull/1403)

This is the only pipeline that publishes whether the pipeline succeeds or fails, so it's the only job vulnerable to duplicate artifacts

### Add additional test for ContainerID parsing [(#1405)](https://github.com/DataDog/dd-trace-dotnet/pull/1405)

Python recently had an issue with container ID parsing in https://github.com/DataDog/dd-trace-py/issues/2314. This adds the problematic ID to our test suite to ensure we are not affected

### Revert "ProcessExit event handler improvements [(#1332)](https://github.com/DataDog/dd-trace-dotnet/pull/1332)" [(#1410)](https://github.com/DataDog/dd-trace-dotnet/pull/1410)

This reverts commit 0aaa30eb253a6af8792cbd1ff5918927c663d777.

### Generate nuget symbols package [(#1401)](https://github.com/DataDog/dd-trace-dotnet/pull/1401)

* Generate nuget symbols package
* Enable source-link
* Enable deterministic builds

### Bump version to 1.26.0 [(#1411)](https://github.com/DataDog/dd-trace-dotnet/pull/1411)

* Bump version to 1.26.0

### Re-enable AspNet integration tests in CI [(#1414)](https://github.com/DataDog/dd-trace-dotnet/pull/1414)

Runs the AspNet tests added by https://github.com/DataDog/dd-trace-dotnet/pull/1288 . Currently, this only tests the `AspNetMvc` and `AspNetWebApi2` integrations because the `AspNet` integration is not yet enabled via automatic instrumentation.

The tests are run in the Windows IIS runs for the following reasons:
- Restoring the projects and publishing can be grouped into the existing `samples-iis.sln` solution
- Instrumenting IIS and IIS Express requires the Datadog.Trace assemblies to be in the GAC

- Adding the AspNet CI applications to `samples-iis.sln`, which we are already properly restoring and publishing in the Windows IIS jobs
- Removing some compile-time references to `Datadog.Trace.dll` and `Datadog.Trace.ClrProfiler.Managed.dll` with lookups via reflection. This is used so the site can give us a clear indicator whether automatic instrumentation is running
- Miscellaneous change: Stop Visual Studio from auto-generating a `launchSettings.json` for the Datadog.Trace.IntegrationTests project

### Ducktype reverse proxy [(#1402)](https://github.com/DataDog/dd-trace-dotnet/pull/1402)

* Add support for reverse proxy/ducktyping.

* Fixes ducktyping for non visible types.

* Changes ReverseProxyTests

* Enables tests on NET451, and adds a new test using public types.
Changes based on the review.

* Apply suggestions from code review

Co-authored-by: Zach Montoya <zach.montoya@datadoghq.com>

* Apply suggestions from code review

Co-authored-by: Kevin Gosse <krix33@gmail.com>

* Changes based on the review.

Co-authored-by: Zach Montoya <zach.montoya@datadoghq.com>
Co-authored-by: Kevin Gosse <krix33@gmail.com>

### Fix git parser on really big pack files (>2GB) [(#1413)](https://github.com/DataDog/dd-trace-dotnet/pull/1413)

* Fix git parser on really big pack files. > 2GB

* Fix parser on big object size.

* changes based in the comments.

* Fixes.

* Apply suggestions from code review

Co-authored-by: Andrew Lock <andrew.lock@datadoghq.com>

Co-authored-by: Andrew Lock <andrew.lock@datadoghq.com>

### Reinstate the consolidated multi-stage build pipeline [(#1363)](https://github.com/DataDog/dd-trace-dotnet/pull/1363)

Re-incorporate the unified build pipeline and ensure that the main pipelines can successfully run in parallel. The main difference between the two is that, for Linux builds, the managed assets are no longer being copied during the Profiler step but instead during the package step. Thus, the only changes in this PR that affects shared scripts are new conditional blocks that check a new `UNIFIED_PIPELINE` environment variable to determine in which step to copy the managed assets.

Notes:
- Reverts the "revert" PR https://github.com/DataDog/dd-trace-dotnet/pull/1335
- Adds the NuGet config change from https://github.com/DataDog/dd-trace-dotnet/pull/1353, which assumes the change will be taken

### Bump version to 1.26.1 [(#1424)](https://github.com/DataDog/dd-trace-dotnet/pull/1424)

* Bump version to 1.26.1

* Update Changelog

### Fix build issue when building MSI from the command line without the TracerHomeDirectory argument [(#1423)](https://github.com/DataDog/dd-trace-dotnet/pull/1423)

### CODE REVIEW

File: Datadog.Trace.proj
Comment (pjanotti) [inline Datadog.Trace.proj:5]: @dszmigielski keep an eye if these will affect the PR to generate the build artifacts.

**Review by pjanotti — COMMENTED** [review]
LGTM - Let's wait on CI completion to see any errors.

File: Datadog.Trace.proj
Comment (dszmigielski) [reply Datadog.Trace.proj:5]: It should not, but I will test it after this one is merged.

**Review by nrcventura — APPROVED** [review]

File: Datadog.Trace.proj
Comment (zacharycmontoya) [reply Datadog.Trace.proj:5]: The `packages` build should be fine because it sets the `TracerHomeDirectory` property itself, but I had to issue a bug fix for local builds if you don't set it: https://github.com/DataDog/dd-trace-dotnet/commit/1acaff04b3d425e4da13bb3745381cf275cbc612

**Review by zacharycmontoya — APPROVED** [review]

**Comment by pjanotti** [comment]
IIRC this is a flaky test - kicked it again to be safe:

```terminal
2021-04-26T23:18:06.9158006Z [xUnit.net 00:15:42.47]     Datadog.Trace.ClrProfiler.IntegrationTests.SmokeTests.DuplicateTypeProxySmokeTest.NoExceptions [FAIL]
2021-04-26T23:18:06.9267184Z   Failed Datadog.Trace.ClrProfiler.IntegrationTests.SmokeTests.DuplicateTypeProxySmokeTest.NoExceptions [25 s]
2021-04-26T23:18:06.9667041Z   Error Message:
2021-04-26T23:18:06.9775798Z    Non-success exit code 139
2021-04-26T23:18:06.9777087Z Expected: True
2021-04-26T23:18:06.9778225Z Actual:   False
2021-04-26T23:18:06.9779277Z   Stack Trace:
2021-04-26T23:18:06.9781226Z      at Datadog.Trace.ClrProfiler.IntegrationTests.SmokeTests.SmokeTestBase.CheckForSmoke(Boolean shouldDeserializeTraces) in /project/test/Datadog.Trace.ClrProfiler.IntegrationTests/SmokeTests/SmokeTestBase.cs:line 47
2021-04-26T23:18:06.9813350Z   Standard Output Messages:
2021-04-26T23:18:06.9815165Z  Application path: /project/test/test-applications/regression/DuplicateTypeProxy/bin/Release/netcoreapp2.1/publish/DuplicateTypeProxy.dll
2021-04-26T23:18:06.9816172Z  Executable path: dotnet
2021-04-26T23:18:06.9816619Z  Assigning port 38605 for the agentPort.
2021-04-26T23:18:06.9817290Z  Assigning port 36411 for the aspNetCorePort.
2021-04-26T23:18:06.9818472Z  Attempt 1: Unable to find profiler at /project/test/test-applications/regression/DuplicateTypeProxy/bin/Release/netcoreapp2.1/publish/profiler-lib/OpenTelemetry.AutoInstrumentation.ClrProfiler.Native.so.
2021-04-26T23:18:06.9820351Z  Attempt 2: Unable to find profiler at /project/test/Datadog.Trace.ClrProfiler.IntegrationTests/bin/Release/netcoreapp2.1/publish/profiler-lib/OpenTelemetry.AutoInstrumentation.ClrProfiler.Native.so.
2021-04-26T23:18:06.9821462Z  Found profiler at /project/src/Datadog.Trace.ClrProfiler.Native/bin/Release/x64/OpenTelemetry.AutoInstrumentation.ClrProfiler.Native.so.
2021-04-26T23:18:06.9827311Z  Found integrations at /project/test/test-applications/regression/DuplicateTypeProxy/bin/Release/netcoreapp2.1/publish/profiler-lib/integrations.json.
2021-04-26T23:18:06.9828388Z  StandardOutput:
2021-04-26T23:18:06.9828855Z  Writing full dump to file /tmp/coredump.6590
2021-04-26T23:18:06.9829536Z  Written 3101057024 bytes (757094 pages) to core file
2021-04-26T23:18:06.9829988Z  
2021-04-26T23:18:06.9830193Z 
2021-04-26T23:18:06.9830399Z 
2021-04-26T23:18:13.2720874Z [xUnit.net 00:15:48.84] Datadog.Trace.ClrProfiler.IntegrationTests: SUCCESS: Datadog.Trace.ClrProfiler.IntegrationTests.Elasticsearch5Tests.SubmitsTraces(5.3.1, True, False) 
```

**Comment by pjanotti** [comment]
Another failure but a different one this time, since 2 runs we got 2 different failures I'm considering both results of flaky tests. I will merge the change.

```terminal
2021-04-27T00:05:10.1092713Z [xUnit.net 00:00:15.78]     Datadog.Trace.IntegrationTests.DiagnosticListeners.AspNetCoreDiagnosticObserverTests.DiagnosticObserver_ForMvcEndpoints_SubmitsSpans(path: "/I/dont/123/exist/", statusCode: 404, isError: False, resourceName: "GET /i/dont/?/exist/", expectedTags: [[aspnet_core.route, ], [aspnet_core.controller, ], [aspnet_core.action, ], [aspnet_core.area, ], [aspnet_core.page, ], ...]) [FAIL]
2021-04-27T00:05:11.0981859Z   Failed Datadog.Trace.IntegrationTests.DiagnosticListeners.AspNetCoreDiagnosticObserverTests.DiagnosticObserver_ForMvcEndpoints_SubmitsSpans(path: "/I/dont/123/exist/", statusCode: 404, isError: False, resourceName: "GET /i/dont/?/exist/", expectedTags: [[aspnet_core.route, ], [aspnet_core.controller, ], [aspnet_core.action, ], [aspnet_core.area, ], [aspnet_core.page, ], ...]) [482 ms]
2021-04-27T00:05:11.0992795Z   Error Message:
2021-04-27T00:05:11.0998349Z    Expected var parentSpan = trace to contain 1 item(s), but found <null>.
2021-04-27T00:05:11.1004622Z   Stack Trace:
2021-04-27T00:05:11.1008963Z      at FluentAssertions.Execution.XUnit2TestFramework.Throw(String message)
2021-04-27T00:05:11.1009912Z    at FluentAssertions.Execution.AssertionScope.FailWith(Func`1 failReasonFunc)
2021-04-27T00:05:11.1013751Z    at FluentAssertions.Collections.SelfReferencingCollectionAssertions`2.HaveCount(Int32 expected, String because, Object[] becauseArgs)
2021-04-27T00:05:11.1020079Z    at Datadog.Trace.IntegrationTests.DiagnosticListeners.AspNetCoreDiagnosticObserverTests.AssertDiagnosticObserverSubmitsSpans[T](String path, HttpStatusCode statusCode, Boolean isError, String resourceName, SerializableDictionary expectedParentSpanTags, Boolean featureFlag, Int32 spanCount, String childSpan1ResourceName, SerializableDictionary firstChildSpanTags, String childSpan2ResourceName, SerializableDictionary secondChildSpanTags) in /project/test/Datadog.Trace.IntegrationTests/DiagnosticListeners/AspNetCoreDiagnosticObserverTests.cs:line 191
2021-04-27T00:05:11.1025162Z    at Datadog.Trace.IntegrationTests.DiagnosticListeners.AspNetCoreDiagnosticObserverTests.DiagnosticObserver_ForMvcEndpoints_SubmitsSpans(String path, HttpStatusCode statusCode, Boolean isError, String resourceName, SerializableDictionary expectedTags) in /project/test/Datadog.Trace.IntegrationTests/DiagnosticListeners/AspNetCoreDiagnosticObserverTests.cs:line 28
2021-04-27T00:05:11.1032076Z --- End of stack trace from previous location where exception was thrown ---
2021-04-27T00:05:20.8680123Z Results File: /project/test/Datadog.Trace.IntegrationTests/results/_c4e9b8c7bab2_2021-04-27_00_05_00.trx
2021-04-27T00:05:20.8684805Z 
```

**Pull request merged**
